### DISCUSSION TITLE
Can LosslessCut automatically close current video after export?

### DISCUSSION THREAD

**Discussion by ALonelyJuicebox** [mifi/lossless-cut#1425] [Q&A]
After editing video, I often find myself doing a quick check to ensure the clip is fine, then deleting the video in Windows Explorer, which will throw an error if the file is still open in LosslessCut.

Is it possible to have LosslessCut auto-close the current file once the export is complete?
If not, I'll submit a feature request.

Thanks!

**Comment by mifi** [comment]
Can’t you just ctrl+w after export is done? Or what is your use case for this

**Comment by ALonelyJuicebox** [comment]
I'm a fan of efficiency where possible! I'd love the ability to skip the step of clicking "Close file" or pressing `Ctrl+W`. Once the export is complete, LosslessCut doesn't have any need to continue working with the file. For my use case (and as an option in Settings) I'd prefer it release the file handle by closing the file automatically. 🙏🏿

**Reply by mifi** [reply]
ok I have now implemented a choice in the cleanup dialog that will allow doing this!

<img width="490" alt="Screenshot 2023-01-27 at 17 02 12" src="https://user-images.githubusercontent.com/402547/215037327-cd5e607b-7cd2-48f7-b565-7f8757d2b7dc.png">

**Reply by Minoo7** [reply]
This does not seem to work when opening a new file after shutting down the lossless cut application. Can we make it always remember to trash original source?

**Reply by mifi** [reply]
Maybe. But i think it’s not *that* inconvenient to have to do it everytime losslesscut has been restarted (unless you restart lossless cut all the time, in which case it would be interesting to know why)

**Comment by ALonelyJuicebox** [comment]
Amazing! 😍
Thank you so much!

**Comment by david-017** [comment]
I don't know if this is the right place or if this is closed. Speaking about this request, I would like it to be fulfilled exactly as the user first requested, auto closing file or, much better still, Don't trash it please! Delete file straight foward! Files in trash still take disk storage, unless it exceeds the trash limit. But the problem is trashing the files then we have to go to empty trash to free disk space really.

currently I manually close the file and then holding the windows key shortcut to delete file permanently without trashing. Would be great if you can change or add the option to delete files permanently. Thanks so much! this app is great and very well maintained!

**Reply by mifi** [reply]
I think trash is a nice way to prevent accidentally permanently deleting files. It gives the users an extra chance to undo their mistake. How much extra work is it to empty trash once every minute or so?

**Reply by snakecase** [reply]
Definitely send it to Recycle Bin only so the LosslessCut doesn't accidentally become LossyCut 😋. It's up to the user to delete their videos. They could also set their Recycle Bin maximum size for each drive separately or disable it altogether.

![explorer_2023-02-07_14-00-33](https://user-images.githubusercontent.com/4579891/217199166-4e513cdf-7d10-41ac-99a6-8b30706722d3.jpg)

**Comment by david-017** [comment]
Ok agreed. Thanks snakecase for pointing that out!

**Comment by ALonelyJuicebox** [comment]
Sorry folks... I must be losing my mind here-- I have the latest build (Windows, LosslessCut v3.50.0) installed-- how do I trigger this feature? I've looked in settings, I've checked the settings that pop up after clicking the Export button, am I missing something obvious? 😅

**Reply by snakecase** [reply]
No it's really convoluted. You have to do it in the trash bin icon but only when a project is opened:
![LosslessCut_2023-02-16_14-07-36](https://user-images.githubusercontent.com/4579891/219319373-c64c3c13-8fed-4233-994b-26cb2482290f.jpg)
These settings reset every time you close the app though.

**Reply by mifi** [reply]
I have now made it so that the setting will be remembered after restart, and put the config inside settings too
